Mathlib/RingTheory/TensorProduct/Basic.lean

```diff
@@ -1126,6 +1126,49 @@ lemma toMatrix_baseChange (f : M₁ →ₗ[R] M₂) (b₁ : Basis ι R M₁) (b
 
 end LinearMap
 
+namespace LinearMap
+
+variable (R A M : Type*) [CommRing R] [CommRing A] [Algebra R A] [AddCommGroup M] [Module R M]
+
+open Module
+open scoped TensorProduct
+
+/-- The natural linear map $A ⊗ (\text{End}_R M) → \text{End}_A (A ⊗ M)$,
+where `M` is an `R`-module, and `A` an `R`-algebra.
+
+See `TensorProductEnd` for the same map, bundled as `A`-algebra homomorphism. -/
+@[simps!]
+noncomputable
+def TensorProductEndₗ : A ⊗[R] (End R M) →ₗ[A] End A (A ⊗[R] M) :=
+  TensorProduct.AlgebraTensorModule.lift <|
+  { toFun := fun a ↦ a • baseChangeHom R A M M
```

Comment:
How about we support `M →ₗ[R] N` and not just `Module.End R M`:
```suggestion
See `TensorProductEnd` for the case `N = M`, bundled as `A`-algebra homomorphism. -/
@[simps!]
noncomputable
def TensorProductₗ (N : Type*) [AddCommGroup N] [Module R N] :
    A ⊗[R] (M →ₗ[R] N) →ₗ[A] A ⊗[R] M →ₗ[A] A ⊗[R] N :=
  TensorProduct.AlgebraTensorModule.lift <|
  { toFun := fun a ↦ a • baseChangeHom R A M N
```

Reply:
(This could probably be a one-liner if `TensorProduct.AlgebraTensorModule.lift` was generalised to be a linear map rather than just a raw function, but I don't think we should tackle this as part of this work.)

Reply:
Thanks, I made the generalization. And yes, let's postpone that other refactor of `TensorProduct.AlgebraTensorModule.lift`.